src/settings.rs

```diff
@@ -36,7 +36,265 @@ pub struct FacetingSettings {
+    /// Use OpenAi's API to generate embeddings
+    /// Depending on hardware, this is a
+    OpenAi(OpenapiEmbedderSettings),
+    /// [Ollama](https://ollama.com/) is a framework for building and running language models locally.
+    Ollama(OllamaEmbedderSettings),
+    /// Supports arbitrary embedders which supply a [REST](https://en.wikipedia.org/wiki/REST) interface
+    REST(GenericRestEmbedderSettings),
+    /// Provide custom embeddings.
+    /// In this case, you must manually update your embeddings when adding, updating, and removing documents to your index.
+    UserProvided(UserProvidedEmbedderSettings),
+}
+
+/// EXPERIMENTAL
+///
+/// # Example
+/// ```
+/// # use meilisearch_sdk::settings::HuggingFaceEmbedderSettings;
+/// let embedder_setting = HuggingFaceEmbedderSettings {
+///   model: Some("BAAI/bge-base-en-v1.5".to_string()),
+///   document_template: Some("A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}".to_string()),
+///   ..Default::default()
+/// };
+/// # let expected = r#"{"model":"BAAI/bge-base-en-v1.5","documentTemplate":"A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}"}"#;
+/// # let expected: HuggingFaceEmbedderSettings = serde_json::from_str(expected).unwrap();
+/// # assert_eq!(embedder_setting, expected);
+/// ```
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct HuggingFaceEmbedderSettings {
+    /// the [BERT embedding model](https://en.wikipedia.org/wiki/BERT_(language_model)) you want to use from [HuggingFace](https://huggingface.co)
+    /// Defaults to `"BAAI/bge-base-en-v1.5"`
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub model: Option<String>,
+    /// revisions allow you to pin a specific version of a model, using a commit hash, tag or branch
+    /// this allows (according to [huggingface](https://huggingface.co/transformers/v4.8.2/model_sharing.html)):
+    /// - built-in versioning
+    /// - access control
+    /// - scalability
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub revision: Option<String>,
+    /// Use it to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+    ///
+    /// if present, `document_template` must be a [Liquid template](https://shopify.github.io/liquid/).
+    /// Use `{{ doc.attribute }}` to access document field values.
+    /// Meilisearch also exposes a `{{ fields }}` array containing one object per document field, which you may access with `{{ field.name }}` and `{{ field.value }}`.
+    ///
+    /// For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.
+    /// Example: "A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
```

Comment:
```suggestion
    /// Example: `"A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"`
    /// 
    /// Default:
    /// ```raw
    /// {% for field in fields %}
    /// {% if field.is_searchable and not field.value == nil %}
    /// {{ field.name }}: {{ field.value }}\n
    /// {% endif %}
    /// {% endfor %}
    /// ```
```